x-pack/legacy/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -198,14 +223,62 @@ export function XYChart({ data, args, formatFactory, timeZone, chartTheme }: XYC
         showLegendExtra={false}
         theme={chartTheme}
         rotation={shouldRotate ? 90 : 0}
-        xDomain={
-          data.dateRange && layers.every(l => l.xScaleType === 'time')
-            ? {
-                min: data.dateRange.fromDate.getTime(),
-                max: data.dateRange.toDate.getTime(),
-              }
-            : undefined
-        }
+        xDomain={xDomain}
+        onElementClick={([[geometry, series]]) => {
+          const xySeries = series as XYChartSeriesIdentifier;
+          const xyGeometry = geometry as GeometryValue;
```

Comment:
Whenever we cast, we should leave some comment in code, why we know that this cast is true, since every cast basically disables the type system, and if this will create later a bug due to no automatic type checks, it's useful to see the thoughts from why this was castable in the past.

I.e. why do we know `geometry` can never be `LayerValue[]` in our case? Why do we know `series` is never `SeriesIdentifier`?